internal/controller/cloud-control/iprange_azure_test.go

```diff
@@ -205,6 +207,36 @@ var _ = Describe("Feature: KCP IpRange for Azure", func() {
 			Expect(azureSecurityGroup).NotTo(BeNil())
 		})
 
+		var virtualNetworkLink *armprivatedns.VirtualNetworkLink
+		resourceGroupName := azurecommon.AzureCloudManagerResourceGroupName(scope.Spec.Scope.Azure.VpcNetwork)
+		privateDnsZoneName := azureutil.NewPrivateDnsZoneName(kcpIpRangeName)
+		By("Then Azure Virtual Network Link is created", func() {
+			Eventually(func() error {
+				vnl, err := azureMock.GetVirtualNetworkLink(infra.Ctx(), resourceGroupName, privateDnsZoneName, kcpIpRangeName)
+				if err != nil {
+					return fmt.Errorf("error loading azure Virtual Network Link: %w", err)
+				}
+				virtualNetworkLink = vnl
+				return nil
+			}).
+				Should(Succeed())
+			Expect(virtualNetworkLink).NotTo(BeNil())
+		})
```

Comment:
In other places where we had a similar check, we would usually add a field to the status that would provide a reference to it.
We would then check that the field is set via ObjAssertion (e.g. `HavingStatusVirtualNetworkLink`).
When it becomes set, we would query the mock outside of the "eventually" block to fetch it.

See example here:
https://github.com/kyma-project/cloud-manager/blob/236bee374e43c2c24999065a3458b93e92b2c660/internal/controller/cloud-control/nfsinstance_aws_test.go#L88-L95

Reply:
No, in the case of Azure we know the reference to dnsZone and virtualPrivateLink based on "well known name" pattern (UUID).

We do not add this ref to KCP CR (as we do for the AWS or CCEE), so I can not use proposed pattern.